core/src/main/java/org/infinispan/stats/impl/AbstractClusterStats.java

```diff
@@ -0,0 +1,177 @@
+package org.infinispan.stats.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.annotations.Start;
+import org.infinispan.jmx.JmxStatisticsExposer;
+import org.infinispan.jmx.annotations.DataType;
+import org.infinispan.jmx.annotations.DisplayType;
+import org.infinispan.jmx.annotations.ManagedAttribute;
+import org.infinispan.jmx.annotations.ManagedOperation;
+import org.infinispan.jmx.annotations.Parameter;
+import org.infinispan.jmx.annotations.Units;
+import org.infinispan.util.TimeService;
+import org.infinispan.util.logging.Log;
+
+/**
+ * @author Ryan Emerson
+ * @since 9.0
+ */
+public abstract class AbstractClusterStats implements JmxStatisticsExposer {
+
+   public static final long DEFAULT_STALE_STATS_THRESHOLD = 3000;
+
+   private TimeService timeService;
+   private volatile long staleStatsThreshold = DEFAULT_STALE_STATS_THRESHOLD;
+   private volatile long statsUpdateTimestamp = 0;
+   volatile boolean statisticsEnabled = false;
+
+   private final Log log;
+   private final AtomicLong resetNanoseconds = new AtomicLong(0);
+   final HashMap<String, Number> statsMap = new HashMap<>();
+
+   AbstractClusterStats(Log log) {
+      this.log = log;
+   }
+
+   abstract void updateStats() throws Exception;
+
+   @Inject
+   void init(TimeService timeService) {
+      this.timeService = timeService;
+   }
+
+   @Start
+   void start() {
+      setStatisticsEnabled(statisticsEnabled);
+   }
+
+   public void reset() {
+      statsMap.clear();
+   }
+
+   @ManagedOperation(description = "Gets the threshold for cluster wide stats refresh (milliseconds)",
+         name = "getStaleStatsThreshold",
+         displayName = "Get the cluster wide stats refresh threshold")
+   public long getStaleStatsThreshold() {
+      return staleStatsThreshold;
+   }
+
+   @ManagedOperation(description = "Sets the treshold for cluster wide stats refresh (milliseconds)",
```

Comment:
We should be able to use 1 Operation definition just like we have with `isStatisticsEnabled`.